src/export/StructureDefinitionExporter.ts

```diff
@@ -408,6 +409,16 @@ export class StructureDefinitionExporter implements Fishable {
     }
 
     const parentName = fshDefinition.parent || 'Resource';
+
+    if (fshDefinition.name === fshDefinition.parent) {
+      const result = this.fishForMetadata(parentName, Type.Resource);
+      throw new ParentDeclaredAsProfileNameError(
+        fshDefinition.name,
+        fshDefinition.sourceInfo,
```

Comment:
Do we actually need to track the `sourceInfo` here? It seems like the `sourceInfo` would get added on line 466 when the error is caught and logged. I'm not sure if there is a case where the `sourceInfo` tracked here would differ from the `sourceInfo` logged there, since it seems like they both just come from the `fshDefinition`.

Reply:
I think you're right that it won't change. I was mirroring what it seems like the `ParentNotDefinedError` below this one does, but I don't think the source info is actually being logged from either of those errors now that I look more closely. Do you have a preference for removing it to use the info logged on line 466 or keep it for consistency's sake?

Reply:
I can't seem to find a reason why it needs to be there, so if you can't either then yeah I think that makes sense to be consistent. Makes me a little nervous because I trust there would have been some reason at the time, but it seems like perhaps times have changed.

Reply:
I agree. Perhaps times have changed. I think I vote to keep the sourceInfo on both for consistency and for the sake of having more information on the errors.